pkg/porter/credentials.go

```diff
@@ -194,3 +206,102 @@ func (p *Porter) GenerateCredentials(opts CredentialOptions) error {
 	}
 	return nil
 }
+
+// Validate validates the args provided Porter's credential show command
+func (o *CredentialShowOptions) Validate(args []string) error {
+	switch len(args) {
+	case 0:
+		return errors.Errorf("no credential name was specified")
+	case 1:
+		o.Name = strings.ToLower(args[0])
+	default:
+		return errors.Errorf("only one positional argument may be specified, the credential name, but multiple were received: %s", args)
+	}
+
+	format, err := printer.ParseFormat(o.RawFormat)
+	if err != nil {
+		return err
+	}
+	o.Format = format
+
+	return nil
+}
+
+// ShowCredential shows the credential set corresponding to the provided name, using
+// the provided printer.PrintOptions for display.
+func (p *Porter) ShowCredential(opts CredentialShowOptions) error {
+	credsDir, err := p.Config.GetCredentialsDir()
+	if err != nil {
+		return errors.Wrap(err, "unable to determine credentials directory")
+	}
+
+	credSet := &credentials.CredentialSet{}
+	data, err := p.Context.FileSystem.ReadFile(filepath.Join(credsDir, fmt.Sprintf("%s.yaml", opts.Name)))
+	if err != nil {
+		return errors.Wrapf(err, "unable to load credential set %s", opts.Name)
+	}
+
+	if err = yaml.Unmarshal(data, credSet); err != nil {
+		return errors.Wrapf(err, "unable to unmarshal credential set %s", opts.Name)
+	}
+
+	switch opts.Format {
+	case printer.FormatJson:
+		return printer.PrintJson(p.Out, credSet)
+	case printer.FormatYaml:
+		return printer.PrintYaml(p.Out, credSet)
+	case printer.FormatTable:
+		var data [][]string
+
+		// Iterate through all CredentialStrategies to build up our data set
+		for _, cs := range credSet.Credentials {
+			// Build a reflected Source of type reflectedStruct, for use below
+			reflectedSource := reflectedStruct{
```

Comment:
I suggest making an exported function that accepts `Source` and returns the string representing the the source type. It will be easier to read and I bet others will want to use it too. This looks like it was hard to figure out! 👍

Reply:
Good call; now broken out into exported function.